File: tests/health_contract_smoke.py
from __future__ import annotations

import os
from pathlib import Path
import sys
from unittest.mock import patch

from fastapi.testclient import TestClient

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
from deploy import aliyun_api as api


with patch.dict(
    os.environ,
    {
        "SURVEYKIT_RELEASE": "backend-1",
        "SURVEYKIT_COMMIT": "abc123",
        "SURVEYKIT_DEPLOYED_AT": "2026-08-23T10:00:00Z",
    },
):
    response = TestClient(api.app).get("/healthz")

assert response.status_code == 200
assert response.headers["cache-control"] == "no-store"
assert response.headers["x-surveykit-service"] == "pptx-report"
payload = response.json()
assert payload["ok"] is True
assert payload["service"] == "pptx-report"
assert payload["release"] == {
    "version": "backend-1",
    "revision": "abc123",
    "deployed_at": "2026-08-23T10:00:00Z",
}
assert payload["capabilities"]["pptx_jobs"] is True
assert payload["capabilities"]["ai_jobs"] is True
assert isinstance(payload["capabilities"]["officecli"], bool)
qualitative_renderer = payload["renderers"]["qualitative_ppt"]
assert qualitative_renderer["preferred_generation_engine"] == "officecli"
assert qualitative_renderer["fallback_generation_engine"] == "python-pptx"
assert qualitative_renderer["generation_engine"] == (
    "officecli" if payload["capabilities"]["officecli"] else "python-pptx"
)
assert qualitative_renderer["officecli_role"] == "primary_enterprise_renderer_and_quality_gate"
assert payload["limits"]["max_upload_bytes"] > 0
assert payload["instance_id"]

print("backend health contract smoke: ok")

File: tests/health-contract-smoke.mjs
import assert from "node:assert/strict";
import { createRequire } from "node:module";
import { onRequestGet } from "../functions/healthz.js";

const require = createRequire(import.meta.url);
const { releaseInfo } = require("../lib/release-info");

const localRelease = releaseInfo({
  SURVEYKIT_RELEASE: "local-1",
  SURVEYKIT_COMMIT: "abc123",
  SURVEYKIT_DEPLOYED_AT: "2026-08-23T10:00:00Z",
});
assert.deepEqual(localRelease, {
  version: "local-1",
  revision: "abc123",
  deployed_at: "2026-08-23T10:00:00Z",
});

const response = await onRequestGet({
  env: {
    PPTX_BACKEND_URL: "https://ppt-api.example.com",
    SENSENOVA_API_KEY: "configured",
    SURVEYKIT_RELEASE: "pages-1",
    CF_PAGES_COMMIT_SHA: "def456",
  },
});
assert.equal(response.status, 200);
assert.equal(response.headers.get("cache-control"), "no-store");
assert.equal(response.headers.get("x-surveykit-service"), "surveykit-web");
const payload = await response.json();
assert.equal(payload.ok, true);
assert.equal(payload.runtime, "cloudflare-pages");
assert.equal(payload.release.version, "pages-1");
assert.equal(payload.release.revision, "def456");
assert.equal(payload.dependencies.pptx_backend_configured, true);
assert.equal(payload.dependencies.ai_proxy_configured, true);

const automaticReleaseResponse = await onRequestGet({
  env: { CF_PAGES_COMMIT_SHA: "commit-release-789" },
});
const automaticReleasePayload = await automaticReleaseResponse.json();
assert.equal(automaticReleasePayload.release.version, "commit-release-789");
assert.equal(automaticReleaseResponse.headers.get("x-surveykit-release"), "commit-release-789");

console.log("Health contract smoke passed: local and Cloudflare release metadata");
